File: Data Structures and Algorithms/Algorithms/selectionSort.py
li=[int(x) for x in input().split()] # input list using comprehension

n=len(li)

for i in range (n-1):# take one element
  minEle=li[i]
  ind=i
  for j in range(i+1,n): 
    # 

    if li[j]<minEle: # compare it with all elements further in the list
      minEle=li[j]
      ind=j
  li[i],li[ind]=minEle,li[i] # replace the element by minimum element found from i+1 to n indices
  
print(li) # print the sorted list

File: Data Structures and Algorithms/Algorithms/c/Selection sort.c
#include<stdio.h>
void selection_sort(int A[],int n)
{
    int i,j,minimum;        
    for(i=0;i<n-1;i++) 
    {
        minimum = i ;
        for(j=i+1;j<n;j++) 
        {
            if(A[j] < A[minimum])  
            {
                minimum = j ;
            }
        }
        int temp = A[minimum];
        A[minimum] = A[i];
        A[i] = temp;
    }
    printf("Sorted array : ");
    for(int i=0;i<n;i++)
    {
        printf("%d ",A[i]);
    }
}
int main()
{
    int N,A[100],i;
    scanf("%d\n",&N);
    for(i=0;i<N;i++)
    {
        scanf("%d",&A[i]);
    }
    selection_sort(A,N);
}
